specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1170 @@
+right-clicked. This would typically be used by developers to ask the user for setting changes.
+
+![Button ContextMenu Example](ButtonContextMenuExample.png)
+
+AppNotificationContent code:
+
+```c#
+AppNotificationContent()
+    .AddText(L"Content")
+    .AddTextBox(L"textBox", L"Reply")
+    .AddButton(Button(L"Modify app settings")
+        .AddArgument(L"action", L"textReply")
+        .UseContextMenuPlacement())
+    .GetXml();
+```
+
+XML output:
+
+```c#
+<toast>
+  <visual>
+    <binding template="ToastGeneric">
+      <text>Content</text>
+    </binding>
+  </visual>
+
+    <actions>
+            <input id="textBox" type="text" placeHolderContent="Reply"/>
+            <action
+            placement="contextMenu"
+            content="Modify app settings"
+            imageUri="Assets/Icons/message.png"
+            arguments="action=textReply"/>
+    </actions>
+</toast>
+```
+
+## Limitations
+
+-   hint-toolTip and hint-buttonStyle are only supported for builds 19041 (20H1) and above. Using
+    these attributes on lower builds will not prevent the AppNotification from being constructed,
+    but developers will receive a debug warning that the attributes are not supported.
+
+# Image
+
+The Image component API sets up the xml for the \<image\> element. Images are visual elements that
+are used to enhance an AppNotification.
+
+The image source can be pulled using one of these protocol handlers:
+
+-   http:// or https:// (A web-based image) (unavailable for unpackageds on build 22000)
+-   ms-appx:/// (An image included in the app package) (only for packaged applications)
+-   ms-appdata:///local/ (An image saved to local storage) (only for packaged applications)
+-   file:/// (A local image)
+
+**WinAppSDK 1.2 \<image\> Schema**:
+
+```c#
+<image src = string
+       alt? = string
+       placement? = "appLogoOverride" | "hero"
+       hint-crop? = "circle" />
+```
+
+Images can be displayed in three different ways:
+
+## Inline
+
+A full-width inline-image that appears when you expand the AppNotification.
+
+![Inline Image Example](InlineImageExample.png)
+
+AppNotificationContent code:
+
+```c#
+AppNotificationContent()
+    .AddText(L"Below is an inline image")
+    .SetInlineImage(winrt::Windows::Foundation::Uri(LR"(https://unsplash.it/360/202?image=883)"))
+    .GetXml();
+```
+
+XML output:
+
+```c#
+<toast>
+  <visual>
+    <binding template="ToastGeneric">
+      <text>Below is an inline image</text>
+      <image src="https://unsplash.it/360/202?image=883">
+    </binding>
+  </visual>
+</toast>
+```
+
+## AppLogoOverride
+
+An image that is left-aligned with notification text.
+
+![AppLogoOverride Image Example](AppLogoOverrideImageExample.png)
+
+AppNotificationContent code:
+
+```c#
+AppNotificationContent()
+    .AddText(L"To the left is an AppLogoOverride")
+    .SetAppLogoOverride(winrt::Windows::Foundation::Uri(LR"(https://unsplash.it/64?image=669)"))
+  .GetXml();
```

Comment:
```suggestion
```cpp
AppNotificationContent()
    .AddText(L"To the left is an AppLogoOverride")
    .SetAppLogoOverride(winrt::Windows::Foundation::Uri(LR"(https://unsplash.it/64?image=669)"))
    .GetXml();
```

Reply:
Fixing indention issues for all occurences.